20 style recipe cardcss part kk

komalgill0310 opened the pull request:
### **What changes did I make? **

1. Gave classnames to the recipe card elements
2. Wrap recipe data in a parent container
3. Imported the font family in the index.css
4.  Styled recipe card. 

### **Why did you make these changes? **

1. Needed to style Recipe card

### **Visual Changes(if applicable)**

1. Recipe card below styles


https://github.com/chingu-voyages/v46-tier2-team-20/assets/80730790/47e97816-66a5-46c4-815a-bdde4d4e3349

### **Known Issue**

1. Some of the images' dimensions are getting changed, due to which there is a white background around the image. 

![image](https://github.com/chingu-voyages/v46-tier2-team-20/assets/80730790/d5313128-a5c2-40c0-8191-b4da8fff740b)

komalgill0310 committed:
committing local changes before pull

komalgill0310 committed:
created a css file for RecipeBrief component

komalgill0310 committed:
added class name for recipe card items

komalgill0310 committed:
1. added recipe image as a background image for the recipe container
2. styled background image

komalgill0310 committed:
styled recipe card

komalgill0310 committed:
convert px value to relative units

komalgill0310 committed:
added flex css style to jsx as a tailwind class for recipe card

robester0403 commented on .eslintcache:
```diff
@@ -0,0 +1 @@
+[{"C:\\Users\\Owner\\Desktop\\DesktopFolder\\Voyages_Projects\\v46-tier2-team-20\\src\\main.jsx":"1","C:\\Users\\Owner\\Desktop\\DesktopFolder\\Voyages_Projects\\v46-tier2-team-20\\src\\App.jsx":"2","C:\\Users\\Owner\\Desktop\\DesktopFolder\\Voyages_Projects\\v46-tier2-team-20\\src\\components\\header\\Header.jsx":"3","C:\\Users\\Owner\\Desktop\\DesktopFolder\\Voyages_Projects\\v46-tier2-team-20\ ...
```
This should be in git ignore

komalgill0310 replied:
moved the file to gitignore

robester0403 commented on src/components/RecipesDisplay/RecipeBrief.css:
```diff
@@ -0,0 +1,82 @@
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+
```
You don't need a space here

robester0403 commented on src/components/RecipesDisplay/RecipeBrief.css:
```diff
@@ -0,0 +1,82 @@
+.recipe-card{
+  width: 300px;
+  height: 225px;
+  margin-bottom: 1.94em; /* 31px */
```
Ideally, your units should be in multiples of 0.25em or multiples of 4px

komalgill0310 replied:
update px to multiples of 4.

robester0403 commented on src/components/RecipesDisplay/RecipeBrief.css:
```diff
@@ -0,0 +1,82 @@
+  transition: transform 0.5s ease, opacity 0.5s ease, box-shadow 0.5s ease; 
+
+  cursor: pointer;
+  margin-bottom: 1.94em; /* 31px */
```
In addition, there's no need for a comment on what the px is.

komalgill0310 replied:
removed comments.

robester0403 commented on src/components/RecipesDisplay/RecipeBrief.css:
```diff
@@ -0,0 +1,82 @@
+  cursor: pointer;
+  margin-bottom: 1.94em; /* 31px */
+
+  /* shadow elevation 1 */
```
it's better to define the variable instead of having it as a comment to what it is in your designs.

komalgill0310 replied:
removed comments.

robester0403 commented on src/components/RecipesDisplay/RecipeBrief.css:
```diff
@@ -0,0 +1,82 @@
+    font-size: 1rem; /* 16px */
+  }
+
+  .recipe-card:hover{
```
you should nest hover in recipe-card. 

.recipe-card {
   ....
    &:hover{
        ....
    }
}

komalgill0310 replied:
update the style as recommended.

robester0403 commented on src/components/RecipesDisplay/RecipeBrief.jsx:
```diff
@@ -1,16 +1,21 @@
-        <img src={thumbnailUrl} />
-        <h1>{name}</h1>
-        {/* Link for Details */}
+      <div key={nanoid()} className="recipe-card bg-no-repeat bg-cover bg-center rounded-lg flex flex-col items-start justify-end" style={{ backgroundImage: `url(${thumbnailUrl})` }}>
```
I assume nanoid is a random key generator. It is not recommended to use a random key generator as a key. You should use some sort of combination of strings from the recipes object or a unique id provided by the backend, which should also be in the recipes object.

komalgill0310 replied:
updated the key with the id from the data as recommended.

robester0403 reviewed (commented):
comments from brief glance at the code.

robester0403 commented on src/components/RecipesDisplay/RecipeBrief.css:
```diff
@@ -0,0 +1,82 @@
+  background-position: center;
+  background:   linear-gradient(180deg, lightgray 50%, #000 73.77%), no-repeat center / cover;
+
+  transition: transform 0.5s ease, opacity 0.5s ease, box-shadow 0.5s ease; 
```
Your transition is not working as expected in the dev app

komalgill0310 committed:
1. Created recipe image as a child of the recipe card
2. Gave class name to the recipe img
3. Deleted background img style from the recipe card

komalgill0310 committed:
styled recipe card layout

komalgill0310 committed:
Style recipe card itself

komalgill0310 committed:
updated key of the recipe card

komalgill0310 committed:
updated style for the recipe name

komalgill0310 committed:
Updated Api key as request limit has reached for the prevvious APi key(it's a temp solution

robester0403 commented on src/components/RecipesDisplay/RecipeBrief.jsx:
```diff
@@ -1,16 +1,22 @@
+import './RecipeBrief.css';
 
 export default function RecipesBrief({ recipes }) {
+  console.log(recipes);
```
remove console.log

robester0403 commented on src/App.jsx:
```diff
@@ -44,7 +45,9 @@ function App() {
       setIsSearching(false);
     } catch (error) {
-      setError(error);
+      console.log('error: ', error);
```
Remove console.log

Emimint commented on src/components/RecipesDisplay/RecipeBrief.css:
Hey @komalgill0310 ! For me it looks great. Here are some minor suggestions for the style, but it is to be discussed with the others as well: 

```
.recipe-img{
  max-width: 342px;
  max-height: 216px;
  box-shadow: rgb(75, 75, 75) 4px 4px 20px;
  transition: all 0.5s;
}

.recipe-img:hover{
  transform: translateY(5px);
} 
```

komalgill0310 replied:
updated the entire style for the Recipe card.

Emimint reviewed (approved):
Looks good for me 👍

komalgill0310 committed:
updated styling for the recipe card

komalgill0310 committed:
1. Fix styling of recipe card
2. Created a Seperate jsx flle for Recipe container

komalgill0310 committed:
convert px to responsive units, moved recipe container styling to its own css file

komalgill0310 committed:
move the recipe img hover style inside the recipe img style

AOA19 commented on .eslintcache:
@komalgill0310 place this file in your gitignore so it doesn't get pushed to the repo

AOA19 commented on .eslintcache~Stashed changes:
@komalgill0310 this file shouldn't be pushed either

komalgill0310 committed:
removed eslint

komalgill0310 committed:
delete stashed changes

komalgill0310 committed:
refactored recipeCard
